### ISSUE TITLE
script triggered through hotkey causes error

### ISSUE THREAD

**Issue by amloewi** [boppreh/keyboard#194]
I have a status bar app (made with the 'rumps' package) to which I'm trying to bind a hotkey. When the hotkey calls a simple function, like an alert, it works fine, but gives this warning:

    python[58226:599749] WARNING: nextEventMatchingMask should only be
    called from the Main Thread! This will throw an exception in the future.

However when hotkey calls a function that triggers a pygame script, which is the point of the app, it produces this error:

    python[58226:599749] pid(58226)/euid(0) is calling TIS/TSM in non-main thread
    environment, ERROR : This is NOT allowed. Please call TIS/TSM in main thread!!!

Is this because of behavior of the hotkeys? Is there a way to call within the main thread?

**Comment by boppreh** [comment]
Yes, this is because the callbacks are executed in a separate thread.

Instead of executing your code in the hotkey's callback, you can have the callback post a message that is then read by the main thread. This way you ensure the code is executed in the correct environment. I don't have experience with the rumps or pygame packages, but if they rely on main threads they probably have some event system you can use.

**Comment by amloewi** [comment]
Thank you! Just as you said, works great.

Minor additional if unrelated issue (and maybe just another misunderstanding), 'shift' is only working for my right shift (on mac), and 'windows+left shift+y' is telling me 'left shift' is not an option (although it's listed in the readme).
